### PR TITLE
Fix #83241: torch.nn.TripletMarginLoss allowed margin less or equal to 0

### LINKED ISSUE #83241: torch.nn.TripletMarginLoss margin can be less than 0
### 🐛 Describe the bug

Parameter _margin_ of torch.nn.TripletMarginLoss is a value greater than 0 on documentation. However, when its value is less than 0, this API also works.
```
import torch
results={}
arg_1 = -7.0
arg_2 = 2
arg_class = torch.nn.TripletMarginLoss(margin=arg_1,p=arg_2)
arg_3_0 = torch.rand([100, 128], dtype=torch.float32)
arg_3_1 = torch.rand([100, 128], dtype=torch.float32)
arg_3_2 = torch.rand([100, 128], dtype=torch.float32)
arg_3 = [arg_3_0,arg_3_1,arg_3_2]
results['res'] = arg_class(*arg_3)
```

### Versions

pytorch: 1.8.1
python: 3.8.3
os: win11

cc @albanD @mruberry @jbschlosser @walterddr @kshitij12345 @saketh-are

### PR DESCRIPTION
**Pull request by goncrust**
Documentation states that the parameter margin of torch.nn.TripletMarginLoss is greater than 0, however any value was being accepted. Also fixed torch.nn.TripletMarginWithDistanceLoss which had the same problem. Added error test input for the new ValueError.

Fixes #83241

### CODE REVIEW

File: torch/nn/functional.py [hunk 1]
```diff
@@ -4680,6 +4680,8 @@ def triplet_margin_loss(
         reduction_enum = _Reduction.legacy_get_enum(size_average, reduce)
     else:
         reduction_enum = _Reduction.get_enum(reduction)
+    if margin <= 0:
+        raise ValueError("margin must be greater than 0")
```
Comment (mikaylagawarecki) [inline torch/nn/functional.py:4684]: nit:
```suggestion
        raise ValueError(f"margin must be greater than 0, got {margin}")
```

Comment (goncrust) [reply torch/nn/functional.py]: Thank you! I also changed that in triplet_margin_with_distance_loss and updated the test.

**Review by mikaylagawarecki — APPROVED** [review]
Thanks!

**Pull request closed without merging**